packages/scaffold-ui/test/modal/w3m-connect-button.test.ts

```diff
@@ -0,0 +1,133 @@
+import { W3mConnectButton } from '../../src/modal/w3m-connect-button'
+import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest'
+import { elementUpdated, fixture } from '@open-wc/testing'
+import { html } from 'lit'
+import { ModalController } from '@reown/appkit-core'
+import { HelpersUtil } from '../utils/HelpersUtil'
+import type { WuiConnectButton } from '@reown/appkit-ui'
+
+describe('W3mConnectButton', () => {
+  beforeEach(() => {
+    ModalController.close()
+    ModalController.setLoading(false)
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  describe('Rendering', () => {
+    it('renders with default props', async () => {
+      const element: W3mConnectButton = await fixture(
+        html`<w3m-connect-button></w3m-connect-button>`
+      )
+      const button = HelpersUtil.getByTestId(element, 'connect-button')
+
+      expect(button).toBeTruthy()
+      expect(button?.textContent?.trim()).toBe('Connect Wallet')
+      expect(button?.getAttribute('size')).toBe('md')
+    })
+
+    it('renders with custom props', async () => {
+      const element: W3mConnectButton = await fixture(
+        html`<w3m-connect-button
+          size="sm"
+          label="Custom Connect"
+          loadingLabel="Custom Loading..."
```

Comment:
is it possible to test also `loadingLabel`?

Reply:
done!